interview-questions.md

```diff
@@ -8,36 +8,167 @@ Then, research the question to expand on your answer. Even if you feel you have
+        -can navigate from the landing page to the view of the new page
+        -can see a form for task and description
+        -can see a sumbit button
+        -can navigate to the landing page from the new page
+
+User Story 4: As a user, I need to be able to create a new task in the ToDo list
+
+Branch: to-do-create
+
+Acceptance Criteria
+-has a controller method for create
+-has a route for create
+-can fill out the form for task and description on the new view
+-the submit button on the new view triggers an action to add the new task to the ToDo database
+-validate the content in the new task, then redirect to the landing page on a successful creation
+
+User Story 5: As a user, I need to see be able to delete a task
+
+Branch: to-do-delete
+
+Acceptance Criteria
+-has a controller method for delete
+-has a route for delete
+-can see a button on a task show page that deletes a task
+-when the delete button is clicked, it performs a delete action
+-after a successful deletion, i am redirected to the landing page.
+
+User Story 6: As a user, I need to see a form to fill out information to edit an existing ToDo task
+
+Branch: to-do-edit
+
+Acceptance Criteria
+-has a controller method for edit
+-has a route for edit
+-has a view for edit
+-can navigate from the show page of an existing task to an edit view of the same task
+-can see a form to edit task and description
+-can see a sumbit button
+-can navigate from the edit view to the original task show page
+-can navigate to the landing page from the edit page
+
+User Story 7: As a user, I need to be able to update an existing task in ToDo
+
+Branch: to-do-update
+
+Acceptance Criteria
+-has a controller for update
+-has a route for update
+-can fill out the form on the task edit page to change the existing data in task and description
+-can click on the submit button to trigger the update action
+-if the update in valid, will be redirected to the show page of the task that has been updated
+
+User Story 8: As a user, I cannot enter invalid data into the database
+
+Branch: to-do-validations
+
+Acceptance Criteria
+-Validate that all ToDo tasks and descriptions are acutal values and are present when initialized
+-Validate that each task entered is unique
+-Validate that each description is at least 5 characters
+
+User Story 9: As a user, I need to be notified when I try to enter invalid data
+
+Branch: to-do-errors
+
+Acceptance Criteria
+-can see a validation error if a ToDo does not include a task or a description
+-can see a validation error if a ToDo is initialized with a duplicate task
+-can see a validation error if a description is less than 5 characters
+-these errors are presented to the user when they perform an incorrect action
+
+User Story 10: As a user, I can see accessible styling that is consistent through each page of the app.
+
+Branch: to-do-styling
+
+Acceptance Criteria
+-has styling applied to each page
+-styling is accessible to users with disabilities
```

Comment:
Outstanding job on these user stories and this will absolutely help you during capstone! 💯